test/integration/projects/projects_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projects
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+type SharedData struct {
+	CloudbuildSA string
+}
+
+func getPolicyID(orgID string, t *testing.T) string {
+	gcOpts := gcloud.WithCommonArgs([]string{"--format", "value(name)"})
+	op := gcloud.Run(t, fmt.Sprintf("access-context-manager policies list --organization=%s ", orgID), gcOpts)
+	return op.String()
+}
+
+func TestProjects(t *testing.T) {
+
+	orgID := utils.ValFromEnv(t, "TF_VAR_org_id")
+	policyID := getPolicyID(orgID, t)
+
+	var sharedData = map[string]*SharedData{
+		"bu1_shared": {
+			CloudbuildSA: "",
+		},
+		"bu2_shared": {
+			CloudbuildSA: "",
+		},
+	}
+
+	for _, tts := range []struct {
+		name  string
+		tfDir string
+	}{
+		{
+			name:  "bu1_shared",
+			tfDir: "../../../4-projects/business_unit_1/shared",
+		},
+		{
+			name:  "bu2_shared",
+			tfDir: "../../../4-projects/business_unit_2/shared",
+		},
+	} {
+		t.Run(tts.name, func(t *testing.T) {
+
+			shared := tft.NewTFBlueprintTest(t,
+				tft.WithTFDir(tts.tfDir),
+			)
+
+			shared.DefineApply(
+				func(assert *assert.Assertions) {
+					// perform default apply of the blueprint
+					shared.DefaultApply(assert)
+					// save the value of the "cloudbuild_sa" to be used in the envs tests
+					sharedData[tts.name].CloudbuildSA = shared.GetStringOutput("cloudbuild_sa")
+				})
+			shared.DefineVerify(
+				func(assert *assert.Assertions) {
+					// perform default verification ensuring Terraform reports no additional changes on an applied blueprint
+					shared.DefaultVerify(assert)
+					// save the value of the "cloudbuild_sa" to be used in the envs tests
+					sharedData[tts.name].CloudbuildSA = shared.GetStringOutput("cloudbuild_sa")
+				})
+			shared.Test()
+		})
+
+	}
+
+	for _, tt := range []struct {
+		name            string
+		sharedData      string
+		perimeterEnvVar string
+		tfDir           string
+		networkTfDir    string
+	}{
+		{
+			name:            "bu1_development",
+			sharedData:      "bu1_shared",
```

Comment:
nit: if we track shared data key as just `bu1` rather than `bu1_shared`, we can split on `bu1_development` to extract just `bu1` and get the shared data.

Reply:
fixed